pkg/syncer/pvcsi_fullsync.go

```diff
@@ -158,11 +161,37 @@ func PvcsiFullSync(ctx context.Context, metadataSyncer *metadataSyncInformer) er
 		log.Errorf("FullSync: Failed to set Guest Cluster labels on SupervisorPVC. Err: %v", err)
 		return err
 	}
-
 	log.Infof("FullSync: End")
 	return nil
 }
 
+// cleanUpCnsVolumeMetadata deletes the cnsvolumemetadata created on legacy kubernetes releases,
+// which are left unused as customer have migrated to non-legacy kubernetes releases
+func cleanUpCnsVolumeMetadata(ctx context.Context, metadataSyncer *metadataSyncInformer,
+	cnsVolumeMetadataList *cnsvolumemetadatav1alpha1.CnsVolumeMetadataList) {
+	log := logger.GetLogger(ctx)
+	log.Info("delete the cnsvolumemetadata created on legacy kubernetes releases")
```

Comment:
log.Info("cleanUpCnsVolumeMetadata: deleting the CnsVolumeMetadata CRs created on legacy vsphere kubernetes releases"):